solr/solrj/src/java/org/apache/solr/client/solrj/request/CollectionAdminRequest.java

```diff
@@ -863,13 +863,23 @@ public SolrParams getParams() {
   }
 
   /**
-   * Return a SolrRequest for low-level detailed status of the collection.
+   * Return a SolrRequest for low-level detailed status of the specified collection. 
+   * @param collection the collection to get the status of.
    */
   public static ColStatus collectionStatus(String collection) {
+    checkNotNull(CoreAdminParams.COLLECTION, collection);
     return new ColStatus(collection);
   }
+  
+  /**
+   * Return a SolrRequest for low-level detailed status of all collections on the cluster.
+   */
+  public static ColStatus collectionStatuses() {
+    return new ColStatus();
+  }
 
-  public static class ColStatus extends AsyncCollectionSpecificAdminRequest {
```

Comment:
AsyncCollectionSpecificAdminRequest enforces a non-null collection name passed in which doesn't work if the solrj colstatus interface's intention is to allow either one collection return or all collections returned as documented in the collection api doc. If no collection = null, i.e. empty, then collectionshandler will treat it as returning all collections staying consistent with the documented api.